datafusion/core/src/datasource/physical_plan/file_scan_config.rs

```diff
@@ -157,6 +157,22 @@ impl FileScanConfig {
         })
     }
 
+    /// Projects only file schema, ignoring partition columns
+    pub(crate) fn projected_file_schema(&self) -> SchemaRef {
```

Comment:
This method looks good, but it would be nice if we could leverage `file_column_projection_indices` (which CsvOpener uses) so we aren't duplicating the logic to exclude the partition columns. 

I think in general we could make Csv, Json, and Arrow file opening / configuring more consistent. We can cut follow on tickets for this.

Reply:
Fixed -- now this method returns schema with fields collected from iteration over `file_column_projection_indices` result